homeassistant/components/overkiz/config_flow.py

```diff
@@ -94,6 +94,8 @@ async def async_step_dhcp(self, discovery_info: dhcp.DhcpServiceInfo) -> FlowRes
 
         _LOGGER.debug("DHCP discovery detected gateway %s", obfuscate_id(gateway_id))
 
+        await self.async_set_unique_id(gateway_id)
+
         if self._gateway_already_configured(gateway_id):
             _LOGGER.debug("Gateway %s is already configured", obfuscate_id(gateway_id))
             return self.async_abort(reason="already_configured")
```

Comment:
```suggestion
        await self.async_set_unique_id(gateway_id)
        self._abort_if_unique_id_configured()
```

Reply:
I will have a look, but I think you are right. It seems that this is part of our legacy code, from before we used gateway as unique id. I don't think it can happen that users have multiple gateways linked to the same Overkiz account/instance.